### DISCUSSION TITLE
documentation of opt_output_files()

### DISCUSSION THREAD

**Discussion by wiesehahn** [r-lidar/lidR#588] [Q&A] [answered]
Hi,
is there any documentation for the options allowed in `opt_output_files(ctg)`?
I have seen many like {ORIGINALFILENAME}, {XBOTTOM},... but where can I see the entire list?

**Comment by Jean-Romain** [comment, marked as answer]
See [this vignette](https://cran.r-project.org/web/packages/lidR/vignettes/lidR-LAScatalog-engine.html#write-independent-chunks-on-disk). It is also documented in `?LAScatalog-class`. It is likely documented in other places but in lidR v4.0.0 I tidied and compacted the documentation and I'm not sure myself where it appears.

There are `XCENTER`, `YCENTER`, `XLEFT`, `YBOTTOM`, `XRIGHT`, `YTOP`, `ID` or `ORIGINALFILENAME`. `ORIGINALFILENAME` is supported only if the chunk size is 0 otherwise it does not have meaning.

You also have `NAMEOFANATTRIBUTE` for functions that support spatial input such as `clip_roi`. See examples in `?clip_roi`
